### PR TITLE
Fix rpm repo to use el in all cases, rather than ansible_distribution

### PR DESCRIPTION
**Pull request by tmgstevens**
Fix rpm repo to use el in all cases, rather than ansible_distribution

### CODE REVIEW

**Comment by gene-redpanda** [comment]
This is not correct for fedora? Not sure why we would want to do this?

```[fedora@ip-10-0-1-64 ~]$ cat /etc/yum.repos.d/redpanda-redpanda.repo 
[redpanda-redpanda]
baseurl = https://dl.redpanda.com/public/redpanda/rpm/fedora/34/$basearch
enabled = 1
gpgcheck = 1
gpgkey = https://dl.redpanda.com/public/redpanda/gpg.988A7B0A4918BC85.key
metadata_expire = 300
name = redpanda-redpanda
proxy = 10.0.2.6:3128
repo_gpgcheck = 1
skip_if_unavailable = 1
sslcacert = /etc/pki/tls/certs/ca-bundle.crt
sslverify = 1
```

**Comment by tmgstevens** [comment]
Here's the results of my investigations so far - firstly the bug I was trying to fix is already fixed here: https://github.com/redpanda-data/redpanda-ansible-collection/commit/f366d916d95e9484bf01b8d70bd9db76c13ee4c1

Secondly, I think we're currently broken on anything that's neither Fedora or RHEL (e.g. Centos, Rocky, Amazon) and in short I don't think there's an easy answer (other than Cloudsmith does this really well with its setup script, which is what we're trying to avoid here).

Here's the ansible facts that we have available to us:

**RHEL**
```
        "ansible_distribution": "RedHat",
        "ansible_distribution_file_parsed": true,
        "ansible_distribution_file_path": "/etc/redhat-release",
        "ansible_distribution_file_search_string": "Red Hat",
        "ansible_distribution_file_variety": "RedHat",
        "ansible_distribution_major_version": "8",
        "ansible_distribution_release": "Ootpa",
        "ansible_distribution_version": "8.8",
        "ansible_os_family": "RedHat",
```
**Amazon**
```
        "ansible_distribution": "Amazon",
        "ansible_distribution_file_parsed": true,
        "ansible_distribution_file_path": "/etc/os-release",
        "ansible_distribution_file_variety": "Amazon",
        "ansible_distribution_major_version": "2",
        "ansible_distribution_minor_version": "NA",
        "ansible_distribution_release": "NA",
        "ansible_distribution_version": "2",
        "ansible_os_family": "RedHat",
```
**Centos**
```
        "ansible_distribution": "CentOS",
        "ansible_distribution_file_parsed": true,
        "ansible_distribution_file_path": "/etc/centos-release",
        "ansible_distribution_file_variety": "CentOS",
        "ansible_distribution_major_version": "8",
        "ansible_distribution_release": "Stream",
        "ansible_distribution_version": "8",
        "ansible_os_family": "RedHat",
```
**Rocky**
```
        "ansible_distribution": "Rocky",
        "ansible_distribution_file_parsed": true,
        "ansible_distribution_file_path": "/etc/redhat-release",
        "ansible_distribution_file_variety": "RedHat",
        "ansible_distribution_major_version": "8",
        "ansible_distribution_release": "Green Obsidian",
        "ansible_distribution_version": "8.7",
        "ansible_os_family": "RedHat",
```
**Fedora**
```
        "ansible_distribution": "Fedora",
        "ansible_distribution_file_parsed": true,
        "ansible_distribution_file_path": "/etc/redhat-release",
        "ansible_distribution_file_variety": "RedHat",
        "ansible_distribution_major_version": "35",
        "ansible_distribution_release": "",
        "ansible_distribution_version": "35",
        "ansible_os_family": "RedHat",
```
And here's what we need to be populating in the YUM repo:
| OS      | Repo Path |
| ----------- | ----------- |
| Centos , Rocky, RHEL, OEL, etc     | el       |
| Fedora   | fedora        |
| Amazon   | amzn        |

So at this point (and given the list of RHEL clones is growing), I'll add code to hard code the cases for Fedora and Amazon and force everything else to go to `el`.

File: roles/redpanda_broker/defaults/main.yml [hunk 1]
```diff
@@ -35,13 +35,18 @@ airgap_copy_dest: "/tmp"
...
-rp_noarch_rpm: "{{ redpanda_base_url }}/public/redpanda/rpm/{{ (ansible_distribution == 'RedHat') | ternary('el', ansible_distribution | lower) }}/{{ ansible_distribution_major_version }}/noarch"
-rp_noarch_rpm_unstable: "{{ redpanda_base_url }}/E4xN1tVe3Xy60GTx/redpanda-unstable/rpm/{{ (ansible_distribution == 'RedHat') | ternary('el', ansible_distribution | lower) }}/{{ ansible_distribution_major_version }}/noarch"
-rp_source_rpm: "{{ redpanda_base_url }}/public/redpanda/rpm/{{ (ansible_distribution == 'RedHat') | ternary('el', ansible_distribution | lower) }}/{{ ansible_distribution_major_version }}/SRPMS"
-rp_source_rpm_unstable: "{{ redpanda_base_url }}/E4xN1tVe3Xy60GTx/redpanda-unstable/rpm/{{ (ansible_distribution == 'RedHat') | ternary('el', ansible_distribution | lower) }}/{{ ansible_distribution_major_version }}/SRPMS"
+rp_standard_rpm: "{{ redpanda_base_url }}/public/redpanda/rpm/{{ rp_rpm_distro_map[ansible_distribution] | default('el') }}/{{ ansible_distribution_major_version }}/$basearch"
+rp_standard_rpm_unstable: "{{ redpanda_base_url }}/E4xN1tVe3Xy60GTx/redpanda-unstable/rpm/el/{{ ansible_distribution_major_version }}/$basearch"
```
Comment (gene-redpanda) [inline roles/redpanda_broker/defaults/main.yml:39]: Just to be sure, this works with unstable on non rhel OS even tho it is forced to el? I'll check tomorrow to be sure, but if you've already tested it it'd be great to know!

Comment (tmgstevens) [reply roles/redpanda_broker/defaults/main.yml]: I didn't, it's a miss, I've fixed it now.

**Review by gene-redpanda — APPROVED** [review]

**Pull request merged**